src/backend/src/lib.rs

```diff
@@ -402,21 +402,38 @@ fn list_user_tokens() -> Vec<Token> {
     read_state(|s| s.user_token.get(&stored_principal).unwrap_or_default().0)
 }
 
-/// Adds a new custom token to the user.
+/// Add, remove or update custom token for the user.
 #[update(guard = "caller_is_not_anonymous")]
-fn add_user_custom_token(token: UserToken) {
+fn set_user_custom_token(token: UserToken) {
     let stored_principal = StoredPrincipal(ic_cdk::caller());
 
-    let find = |t: &UserToken| -> bool { t.clone() == token };
+    let find = |t: &UserToken| -> bool {
+        UserTokenId::from(t.token.clone()) == UserTokenId::from(token.token.clone())
```

Comment:
If the implementation of `From` takes a reference rather than the token itself there is no need to clone.  You can do: `UserTokenId::from(&t.token)`.  I have added a comment  by the `From` implementation with details.